### PR TITLE
Add `thisnode.thisdcos.directory` dns zone

### PR DESCRIPTION
**Pull request by urbanserj**
JIRA: https://jira.mesosphere.com/browse/DCOS_OSS-4666
JIRA: https://jira.mesosphere.com/browse/DCOS-46999

### CODE REVIEW

File: apps/dcos_dns/src/dcos_dns.erl [hunk 1]
```diff
@@ -78,3 +92,109 @@ imp_resolve(DNSName, Family, Timeout) ->
...
+                {error, Error} ->
+                    lager:error(
+                        "Failed to push DNS Zone \"~s\": ~p",
+                        [ZoneName, Error]),
+                    {error, Error}
+            end
+    catch error:Error ->
+        lager:error(
+            "Failed to push DNS Zone \"~s\": ~p",
+            [ZoneName, Error]),
```
Comment (GoelDeepak) [inline apps/dcos_dns/src/dcos_dns.erl]: Do we have a line number in our error logs? If not, then probably we might want to have a slightly different error message just to differentiate from the error message at lineno. 185

**Comment by GoelDeepak** [comment]
What is the purpose of `distance` change?

File: apps/dcos_dns/src/dcos_dns.erl (hunk 1 above)
Comment (urbanserj) [reply apps/dcos_dns/src/dcos_dns.erl]: There is only one possible error with `erldns_zone_cache:get_zone_with_records/1`. It's `error(badarg)`, when erldns cache is not initialized and that's only possible in tests.

Comment (GoelDeepak) [reply apps/dcos_dns/src/dcos_dns.erl]: 👍

**Comment by urbanserj** [comment]
@GoelDeepak I've moved `distance` to new branch [1]. it's a small change, but it's definitely out of the scope for this PR. We can discuss it separately later.

https://github.com/dcos/dcos-net/compare/urbanserj/nodes-distance

**Comment by GoelDeepak** [comment]
> @GoelDeepak I've moved `distance` to new branch [1]. it's a small change, but it's definitely out of the scope for this PR. We can discuss it separately later.
> 
> https://github.com/dcos/dcos-net/compare/urbanserj/nodes-distance

Make sense. Thank you for splitting into two different PRs. It will give us more time to brainstorm the `distance` approach

**Pull request merged**